File: Leetcode Challenge/June/Python/Remove All Adjacent Duplicates In String.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        st = []
        ans=''
        for i in range(len(s)):
            if len(st) == 0:
                st.append(s[i])
            
            elif st[-1] == s[i]:
                st.pop()
            
            else:
                st.append(s[i])
                
        
        while len(st)>0:
            ans += st[-1]
            st.pop()
        return ans[::-1]
            
        

File: Leetcode Challenge/June/Remove All Adjacent Duplicates In String.cpp
class Solution {
public:
    string removeDuplicates(string s) {
        stack<char>st;
        string ans;
        for(int i=0;i<s.size();i++){
            if(st.size() == 0){
                
                st.push(s[i]);
            }
            else if(s[i] == st.top()){
                st.pop();
            }
            else{
                st.push(s[i]);
            }
        }
        
        while(st.size() != 0){
            ans += st.top();
            st.pop();
        }
        
        reverse(ans.begin(),ans.end());
        return ans;
            
        }
};
